### PR TITLE
Implement RBAC decorators and functionality

### PR DESCRIPTION
**Pull request by davidjiglesias**
Hi team,

This PR is related to https://github.com/wazuh/wazuh/issues/3373.

**Unit test results:**
```
============================================================= test session starts =============================================================
platform linux -- Python 3.7.2, pytest-4.5.0, py-1.8.0, pluggy-0.12.0 -- /var/ossec/framework/python/bin/python3.7
cachedir: .pytest_cache
rootdir: /var/ossec/framework/python/bin
plugins: cov-2.7.1, tavern-0.26.4
collected 17 items                                                                                                                            

unit/test_rbac.py::test_get_required_permissions[mock_file1.xml-mock_resources:name:{name}-mock_actions0] PASSED                        [  5%]
unit/test_rbac.py::test_get_required_permissions[mock_file1.xml-mock_resources:name:mock_name-mock_actions0] PASSED                     [ 11%]
unit/test_rbac.py::test_get_required_permissions[mock_names1-mock_resources:name:{name}-mock_actions0] PASSED                           [ 17%]
unit/test_rbac.py::test_get_required_permissions[mock_names1-mock_resources:name:mock_name-mock_actions0] PASSED                        [ 23%]
unit/test_rbac.py::test_get_required_permissions_exception PASSED                                                                       [ 29%]
unit/test_rbac.py::test_match_pairs[False-mock_user0-mock_req0] PASSED                                                                  [ 35%]
unit/test_rbac.py::test_match_pairs[False-mock_user1-mock_req0] PASSED                                                                  [ 41%]
unit/test_rbac.py::test_match_pairs[False-mock_user2-mock_req0] PASSED                                                                  [ 47%]
unit/test_rbac.py::test_match_pairs[True-mock_user0-mock_req0] PASSED                                                                   [ 52%]
unit/test_rbac.py::test_match_pairs[True-mock_user1-mock_req0] PASSED                                                                   [ 58%]
unit/test_rbac.py::test_match_pairs[True-mock_user2-mock_req0] PASSED                                                                   [ 64%]
unit/test_rbac.py::test_matches_privileges[mocked_rbac0] PASSED                                                                         [ 70%]
unit/test_rbac.py::test_matches_privileges[mocked_rbac1] PASSED                                                                         [ 76%]
unit/test_rbac.py::test_matches_privileges[mocked_rbac2] PASSED                                                                         [ 82%]
unit/test_rbac.py::test_matches_privileges[mocked_rbac3] PASSED                                                                         [ 88%]
unit/test_rbac.py::test_get_groups_resources[001] PASSED                                                                                [ 94%]
unit/test_rbac.py::test_get_groups_resources[*] PASSED                                                                                  [100%]

========================================================== 17 passed in 0.08 seconds ==========================================================
```
**Coverage:**
```
----------------------------------------------------------------------------------------------------------------------------
/var/ossec/framework/python/lib/python3.7/site-packages/wazuh-3.10.0-py3.7.egg/wazuh/rbac.py      77      0   100%
```

### CODE REVIEW

**Comment by davidjiglesias** [comment]
**Updated tests results:**
```
============================= test session starts ==============================
platform linux -- Python 3.6.7, pytest-4.5.0, py-1.8.0, pluggy-0.12.0 -- /usr/bin/python3.6
cachedir: .pytest_cache
rootdir: /home/davidjiglesias/git/wazuh/framework
plugins: cov-2.7.1, tavern-0.26.4
collecting ... collected 17 items

test_decorators.py::test_get_required_permissions[mock_file1.xml-mock_resources:name:{name}-mock_actions0] PASSED [  5%]
test_decorators.py::test_get_required_permissions[mock_file1.xml-mock_resources:name:mock_name-mock_actions0] PASSED [ 11%]
test_decorators.py::test_get_required_permissions[mock_names1-mock_resources:name:{name}-mock_actions0] PASSED [ 17%]
test_decorators.py::test_get_required_permissions[mock_names1-mock_resources:name:mock_name-mock_actions0] PASSED [ 23%]
test_decorators.py::test_get_required_permissions_exception PASSED       [ 29%]
test_decorators.py::test_match_pairs[False-mock_user0-mock_req0] PASSED  [ 35%]
test_decorators.py::test_match_pairs[False-mock_user1-mock_req0] PASSED  [ 41%]
test_decorators.py::test_match_pairs[False-mock_user2-mock_req0] PASSED  [ 47%]
test_decorators.py::test_match_pairs[True-mock_user0-mock_req0] PASSED   [ 52%]
test_decorators.py::test_match_pairs[True-mock_user1-mock_req0] PASSED   [ 58%]
test_decorators.py::test_match_pairs[True-mock_user2-mock_req0] PASSED   [ 64%]
test_decorators.py::test_matches_privileges[mocked_rbac0] PASSED         [ 70%]
test_decorators.py::test_matches_privileges[mocked_rbac1] PASSED         [ 76%]
test_decorators.py::test_matches_privileges[mocked_rbac2] PASSED         [ 82%]
test_decorators.py::test_matches_privileges[mocked_rbac3] PASSED         [ 88%]
test_decorators.py::test_get_groups_resources[001] PASSED                [ 94%]
test_decorators.py::test_get_groups_resources[*] PASSED                  [100%]

========================== 17 passed in 0.07 seconds ===========================
```

File: framework/wazuh/database.py [hunk 1]
```diff
@@ -20,7 +20,7 @@
     msg = str(sqlite3.sqlite_version)
     msg += "\nTry to export the internal SQLite library:"
     msg += "\nexport LD_LIBRARY_PATH=$LD_LIBRARY_PATH:{0}/framework/lib".format(common.ossec_path)
-    raise WazuhInternalError(2001, msg)
+    raise WazuhException(2001, msg)
```
Comment (crd1985) [inline framework/wazuh/database.py]:
```suggestion
    raise WazuhException(2001, extra_message=msg)
```

File: framework/wazuh/database.py [hunk 2]
```diff
@@ -90,14 +90,20 @@ def execute(self, query, *args):
                 raise WazuhError(2003, extra_message=str(e))
 
             if n_attempts > self.max_attempts:
-                raise WazuhInternalError(2002, error_text)
+                raise WazuhException(2002, error_text)
```
Comment (crd1985) [inline framework/wazuh/database.py]:
```suggestion
                raise WazuhException(2002, extra_message=error_text)
```

File: framework/wazuh/rbac/decorators.py [hunk 3]
```diff
@@ -0,0 +1,143 @@
+# Copyright (C) 2015-2019, Wazuh Inc.
+# Created by Wazuh, Inc. <info@wazuh.com>.
+# This program is a free software; you can redistribute it and/or modify it under the terms of GPLv2
+
+from functools import wraps
```
Comment (crd1985) [inline framework/wazuh/rbac/decorators.py]: Please, sort imports

File: framework/wazuh/database.py (hunk 2 above)
Comment (crd1985) [inline framework/wazuh/database.py]:
```suggestion
                raise WazuhInternalError(2002, extra_message=error_text)
```

File: framework/wazuh/database.py (hunk 1 above)
Comment (crd1985) [inline framework/wazuh/database.py]:
```suggestion
    raise WazuhInternalError(2001, extra_message=msg)
```

File: framework/wazuh/database.py [hunk 4]
```diff
@@ -35,7 +35,7 @@ def __init__(self, db_path=common.database_path_global, busy_sleep=0.001, max_at
         self.db_path = db_path
 
         if not isfile(db_path):
-            raise WazuhInternalError(2000)
+            raise WazuhException(2000)
```
Comment (crd1985) [inline framework/wazuh/database.py]:
```suggestion
            raise WazuhInternalError(2000)
```

File: framework/wazuh/database.py [hunk 5]
```diff
@@ -5,7 +5,7 @@
 # This program is a free software; you can redistribute it and/or modify it under the terms of GPLv2
 
 from wazuh import common
-from wazuh.exception import WazuhError, WazuhInternalError
+from wazuh.exception import WazuhException, WazuhError, WazuhInternalError
```
Comment (crd1985) [inline framework/wazuh/database.py]:
```suggestion
from wazuh.exception import WazuhError, WazuhInternalError
```

**Pull request merged**